packers/zkp.go

```diff
@@ -238,32 +263,44 @@ func (p *ZKPPacker) MediaType() iden3comm.MediaType {
 }
 
 // DefaultZKPUnpackerOption is a function that sets the default ZKP unpacker options
-type DefaultZKPUnpackerOption func(*defaultZKPUnpacker)
+type DefaultZKPUnpackerOption func(*zkUnpackerOpts)
 
 // WithAuthVerifyDelay sets the delay for the auth verification
 func WithAuthVerifyDelay(delay time.Duration) DefaultZKPUnpackerOption {
-	return func(p *defaultZKPUnpacker) {
+	return func(p *zkUnpackerOpts) {
 		p.authVerifyDelay = delay
 	}
 }
 
-type defaultZKPUnpacker struct {
-	resolvers       map[int]eth.Resolver
+type zkUnpackerOpts struct {
 	authVerifyDelay time.Duration
 }
 
+type defaultZKPUnpacker struct {
+	resolvers map[int]eth.Resolver
+	opts      zkUnpackerOpts
+}
+
 // DefaultZKPUnpacker creates a default ZKP unpacker with the provided verification key and resolvers
 func DefaultZKPUnpacker(verificationKey []byte, resolvers map[int]eth.Resolver, opts ...DefaultZKPUnpackerOption) *ZKPPacker {
-	def := &defaultZKPUnpacker{resolvers, time.Minute * 5}
+	def := &defaultZKPUnpacker{
+		resolvers: resolvers,
+		opts:      zkUnpackerOpts{authVerifyDelay: time.Minute * 5},
+	}
 	for _, opt := range opts {
-		opt(def)
+		opt(&def.opts)
 	}
 	verifications := make(map[jwz.ProvingMethodAlg]VerificationParams)
 	verifications[jwz.AuthV2Groth16Alg] = NewVerificationParams(verificationKey, def.defaultZkpUnpackerVerificationFn)
 	return NewZKPPacker(nil, verifications)
 }
 
-func (d *defaultZKPUnpacker) defaultZkpUnpackerVerificationFn(id circuits.CircuitID, pubsignals []string) error {
+func (d *defaultZKPUnpacker) defaultZkpUnpackerVerificationFn(id circuits.CircuitID, pubsignals []string, opts ...DefaultZKPUnpackerOption) error {
```

Comment:
If you will remove opts from defaultZKPUnpacker, you can refactor the method to function.

Reply:
I don't think so. It is using d.resolvers inside the method body.